### ISSUE TITLE
Management for multiple lists

### ISSUE THREAD

**Issue by bombsimon** [fippli/terminal-todo#12]
It would be nice with some kind of management of multiple lists (text files), in the minimal form print the current list in the header.

Should probably only be implemented if possible with zero/few dependencies. I like the minimalistic implementation/KISS in the current version.

**Comment by fippli** [comment]
Yep, anything that is useful. I think as long as the default behaviour is the simplest idea that just a todo list opens up when running `todo`, we can add all the other options as flags.

How about `todo --lists` opens the "TODO" interface but with different TODO files e.g.
```
[1] todo1
[2] todo2
[3] todo3
...
--------------------------
SELECT TODO LIST: 
```

**Comment by bombsimon** [comment]
Sounds good and I also like the idea to just run `todo`.

I think it would be a good idea to maybe stor all todo-files in `~/.todo/` and name them. Then you could have `~/.todo/default` and `~/.todo/my-custom`. This way you can also avoid to add `.todo` to `.gitignore` (globally or per project).

One solution to the implementation of the "local" `.todo` could be to create a file with the ~~md5-sum~~ base64 encoding of the current directory.

---

1. Create named list, stored as `~/.todo/list-one`

```sh
todo --new 'list-one'
```

2. Create new list for current folder that will auto open, stored as `~/.todo/$(pwd | base64)-default`

```sh
todo --new # `Creates ~/.todo/$(pwd | base64-default)`
```

3. Run `todo`, check if `~/.todo/$(pwd | base64)` exist. If not, open `~/.todo/default`

```sh
todo
```

Open a named list created from step 1. Or maybe create it if not exist and remove step 1?

```sh
todo list-one
```

Create `~/.todo/$(pwd | base64-default)` pre-filled with all TODOs from files in the current directory.

```sh
todo --this-dir
```

Then just use the file names to display the list, or base64 decode the path if it's for a directory.

I also think it would be nice to have this as an option in the menu to switch between lists without restarting the application.

**Comment by fippli** [comment]
Sounds great! I like the idea of storing all the files in the same directory and creating a hash for the filepath as the name of the todo file, or just access it through a specified name.

If the list of files are open in the application, how will their hashed filenames be displayed in the interface? Its a little bit hard to remember a hash right? Or should they just be hidden and available only if you are in that specific directory?

**Comment by bombsimon** [comment]
I chose base64 as encoding (instead of actual hash like md5) to be able to reverse it:

```sh
$ for f in test-one another-test final-test; do touch $(echo $f | base64); done
```

```sh
$ ls -l
total 0
-rw-r--r--  1 simon  staff  0 Dec 11 11:32 YW5vdGhlci10ZXN0Cg==
-rw-r--r--  1 simon  staff  0 Dec 11 11:32 ZmluYWwtdGVzdAo=
-rw-r--r--  1 simon  staff  0 Dec 11 11:32 dGVzdC1vbmUK
```

```sh
$ for f in *; do echo $f | base64 -d; done
another-test
final-test
test-one
```

So yeah, you can just read all the files and decode the base64 encoding :)

Sadly, the bundled `base64` binary up until macOS Catalina was an old version requiring the `-D` flag instead of `-d`. We could check if `base64` supports `-d` with something like `base64 --help | grep -- '-Dd' > /dev/null` (`$?` will be 0 if it does and 1 otherwise).
